feat(cli): Add channel worker settings reload for serve --channel (#6598)
QwenLM/qwen-code: docs/developers/daemon/15-channel-adapters.md

- `shutdown()` closes every active session and the underlying transport (the channel's WebSocket / long-poll).
- DingTalk's WebSocket stream supports server-push; WeChat's long-poll requires a backoff strategy on idle responses; Telegram's long-poll has a built-in `timeout` parameter.

### Settings reload (`POST /workspace/channel/reload`)

The daemon reads channel settings from `settings.json` once, when the channel worker starts (`packages/cli/src/commands/channel/daemon-worker.ts` → `loadSettings` → `loadChannelsConfig`). To apply changes without a full daemon restart, the daemon exposes `POST /workspace/channel/reload` (strict mutation gate; SDK `DaemonClient.reloadChannelWorker()`; CLI `qwen channel reload`):

> 

Reviewer: This documents a failure response that includes the latest worker snapshot, but the route catch path currently delegates to `sendBridgeError(res, err, ...)`, which returns the generic error body and does not include `worker`/snapshot data. Please either include the snapshot in the 5xx response or narrow this doc to say callers should use `GET /daemon/status` for the latest snapshot after a failed reload.

- The route calls `ChannelWorkerSupervisor.restart()` (`packages/cli/src/serve/channel-worker-supervisor.ts`), which stops the current worker child and relaunches it. The relaunched worker re-reads `settings.json`, so channel tokens, `proxy`, and per-channel `model` all take effect.
- Concurrent reloads coalesce onto a single stop+relaunch. `restart()` also resets the crash-restart budget, so a worker parked in `failed` recovers on an explicit reload.

> - If the relaunch fails (for example, settings were edited into an invalid state), the channels stay down, the route returns 5xx with the latest snapshot, and `GET /daemon/status` reports `failed`.

Reviewer 2: **[Suggestion]** The implementation and route test currently return only the bridge error body on relaunch failure (`sendBridgeError` maps this to `{ error: 'relaunch failed' }`), not the latest worker snapshot. This wording will make SDK/HTTP clients expect diagnostic data that the API does not provide. Either include the snapshot in the failure response, or narrow the docs to say the latest state is available from `GET /daemon/status`.

```suggestion
- If the relaunch fails (for example, settings were edited into an invalid state), the channels stay down, the route returns 5xx, and `GET /daemon/status` reports the latest worker state as `failed`.
```

_— GPT-5 via Qwen Code /review_
